### PR TITLE
Add missing type annotations to label transform apply_transform methods

### LINKED ISSUE #1452: apply_transform missing type annotations in 4 label transforms
### Bug summary

Four label transforms override `apply_transform` without type annotations on the `subject` parameter or return type. Every other transform in the codebase uses `subject: Subject) -> Subject:`, so these four are inconsistent and the type checker (`ty`) cannot verify their signatures.

### Code for reproduction

```python
import inspect
import torchio as tio

for cls in [tio.Contour, tio.OneHot, tio.RemapLabels, tio.SequentialLabels]:
    sig = inspect.signature(cls.apply_transform)
    print(cls.__name__, sig)
```

### Actual outcome

```
Contour (self, subject)
OneHot (self, subject)
RemapLabels (self, subject)
SequentialLabels (self, subject)
```

No annotations on `subject` or the return type.

### Expected outcome

Consistent with all other transforms, e.g. `Resample.apply_transform`:

```
(self, subject: Subject) -> Subject
```

### Error messages

No runtime error — this is a type-consistency issue caught by static analysis.

### System info

N/A — this is a source-level inconsistency, not environment-dependent.

### PR DESCRIPTION
**Pull request by rhysmcneill**
Fixes #1452.

**Description**

This is a code quality fix. Four label transforms — `Contour`, `OneHot`, `RemapLabels`, and `SequentialLabels` — override `apply_transform` without type annotations on the `subject` parameter or return type. Every other transform in the codebase already uses `(self, subject: Subject) -> Subject`, making these four inconsistent and invisible to the `ty` static type checker configured in CI.

The fix adds the `Subject` import and the correct annotations to all four methods, bringing them in line with the established pattern across the rest of the transform hierarchy.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/TorchIO-project/torchio/blob/main/CONTRIBUTING.md) docs and have a developer setup ready
- Changes are
  - [x] Non-breaking (would not break existing functionality)
  - [ ] Breaking (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [x] In-line docstrings updated
- [x] Documentation updated
- [x] This pull request is ready to be reviewed

### CODE REVIEW

**Comment by fepegar** [comment]
Thank you, @rhysmcneill!

FYI I've been toying with a re-write of this library. If you have any feature requests or feedback, please let me know.
